src/mocompiler.js

```diff
@@ -6,7 +6,7 @@ import contentType from 'content-type';
  * Exposes general compiler function. Takes a translation
  * object as a parameter and returns binary MO object
  *
- * @param {import('./index.d.ts').GetTextTranslations} table Translation object
+ * @param {import('./types.js').GetTextTranslations} table Translation object
```

Comment:
I would import these at the top of the file, with the other imports then just the types (without the import statement) are used in the function signature. 

I also see it used in quite a few locations in the file, it would look cleaner to not reimport it over and over.

```js
// At the top of the file.
/**
 * @typedef {import('./types.js').GetTextTranslations} GetTextTranslations
 */
```

```suggestion
 * @param {GetTextTranslations} table Translation object
```

Reply:
ref https://github.com/smhg/gettext-parser/pull/89#discussion_r1597472208